File: api/src/main/java/work/beltran/discogsbrowser/api/network/login/RequestHeader.java
package work.beltran.discogsbrowser.api.network.login;

/**
 * Created by Miquel Beltran on 14.05.16.
 * More on http://beltran.work
 */
public class RequestHeader extends LoginHeader {
    private static final String CALLBACK = "discogs://callback";

    public RequestHeader(String consumerKey, String consumerSecret) {
        super(consumerKey, consumerSecret + "&");
    }

    @Override
    public String getHeader() {
        return super.getHeader() + ", " + "oauth_callback=\"" + CALLBACK +"\"";
    }
}

File: business/src/main/java/work/beltran/discogsbrowser/business/login/RequestHeader.kt
package work.beltran.discogsbrowser.business.login

/**
 * Created by Miquel Beltran on 14.05.16.
 * More on http://beltran.work
 */
class RequestHeader(consumerKey: String,
                    consumerSecret: String)
    : LoginHeader(consumerKey, consumerSecret + "&") {

    override fun getHeader(): String {
        return super.header + ", oauth_callback=\"$CALLBACK\""
    }

    companion object {
        private const val CALLBACK = "discogs://callback"
    }
}
